### PR TITLE
Remove `without_storage_info` from pallet `transaction-storage`

### PR DESCRIPTION
**Pull request by skunert**
- Introduces `MaxBlockTransactions` and `MaxTransactionSize` config values
- Removes `MaxBlockTransactions` and `MaxTransactionSize` from storage
- Values remain the same, logic is untouched
- Adds migration to remove the unused storage values

### CODE REVIEW

File: frame/transaction-storage/src/migration.rs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+use frame_support::traits::{Get, PalletInfoAccess, StorageVersion};
+
+use crate::*;
+
+pub struct RemoveLimitsFromStorage<T: crate::Config>(sp_std::marker::PhantomData<T>);
```
Comment (skunert) [inline frame/transaction-storage/src/migration.rs:5]: Is this the preferred way for storage migrations? I was looking around and saw different methods used for different pallets 😬

File: frame/transaction-storage/src/lib.rs [hunk 2]
```diff
@@ -95,6 +105,10 @@ pub mod pallet {
 		type FeeDestination: OnUnbalanced<NegativeImbalanceOf<Self>>;
 		/// Weight information for extrinsics in this pallet.
 		type WeightInfo: WeightInfo;
+
+		type MaxBlockTransactions: Get<u32>;
+
+		type MaxTransactionSize: Get<u32>;
```
[the remark this answers is not in the document]
Comment (skunert) [reply frame/transaction-storage/src/lib.rs]: Sure!

File: frame/transaction-storage/src/migration.rs (hunk 1 above)
Comment (arkpar) [reply frame/transaction-storage/src/migration.rs:5]: I can't answer you question, but I don't think migration is even needed here. There are still no deployed chains that use this crate as far as I know.

Comment (shawntabrizi) [reply frame/transaction-storage/src/migration.rs:5]: yes, migration is not needed

File: frame/transaction-storage/src/benchmarking.rs [hunk 3]
```diff
@@ -152,10 +152,10 @@ benchmarks! {
 		run_to_block::<T>(1u32.into());
 		let caller: T::AccountId = whitelisted_caller();
 		T::Currency::make_free_balance_be(&caller, BalanceOf::<T>::max_value());
-		for _ in 0 .. MaxBlockTransactions::<T>::get() {
+		for _ in 0 .. <T>::MaxBlockTransactions::get() {
```
Comment (shawntabrizi) [inline frame/transaction-storage/src/benchmarking.rs:155]:
```suggestion
		for _ in 0 .. T::MaxBlockTransactions::get() {
```

bit of a weird syntax here, but whatever.

**Review by shawntabrizi — APPROVED** [review]

**Review by davxy — APPROVED** [review]
(Nit) removal of unused `log` crate

**Comment by shawntabrizi** [comment]
bot merge

**Pull request merged**